examples/rustapi_module/tests/test_datetime.py

```diff
@@ -80,8 +83,11 @@ def test_invalid_date_fails():
         rdt.make_date(2017, 2, 30)
 
 
-@given(d=dates())
+@given(d=st.dates())
 def test_date_from_timestamp(d):
+    if PYPY and d < pdt.date(1900, 1, 1):
+        pytest.xfail("get_timestamp will raise on PyPy with dates before 1900")
```

Comment:
I'm not sure if this is hypothesis messing with things, but I'm not actually seeing any `xfail`s in the test suite.

I looked into this to see if it was a bug in PyPy, and basically all the PyPy C API stuff is thin wrappers around calls to Python itself, and at least on Linux, PyPy *does* support large negative values in` fromtimestamp`.

I think this may actually be platform dependent and may not have anything to do with PyPy. @omerbenamram what platform are you seeing the failures for? Windows? Mac?

Reply:
Mac

Reply:
```
latform darwin -- Python 3.5.3[pypy-7.1.0-alpha], pytest-4.3.0, py-1.8.0, pluggy-0.9.0
benchmark: 3.2.2 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=True warmup_iterations=100000)
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/omerba/Workspace/pyo3/examples/rustapi_module/.hypothesis/examples')
rootdir: /Users/omerba/Workspace/pyo3/examples/rustapi_module, inifile:
plugins: faulthandler-1.5.0, benchmark-3.2.2, hypothesis-4.9.0
collected 60 items

tests/test_datetime.py ..XFalsifying example: test_date_from_timestamp(d=datetime.date(1899, 12, 31))
x.ssxxxxXXXXXXX........sX.Falsifying example: test_datetime_from_timestamp(dt=datetime.datetime(1899, 1, 1, 0, 0))
x.......................
tests/test_dict_iter.py    1/  64 iterations:0=>0
```

Reply:
However on cpython (3.7)
```
platform darwin -- Python 3.7.1, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
benchmark: 3.2.2 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/omerba/Workspace/pyo3/examples/rustapi_module/.hypothesis/examples')
rootdir: /Users/omerba/Workspace/pyo3/examples/rustapi_module, inifile:
plugins: benchmark-3.2.2, hypothesis-4.14.0
collected 60 items

tests/test_datetime.py .......xxxx..........................................
```

Reply:
@omerbenamram The ones with the big X are xpasses, so evidently we haven't quite got it right. I'm not seeing either xpasses *or* xfails on PyPy on Travis, though, which is very puzzling to me.

Reply:
you can see it's printing `Falsifying example` which I believe it's hypothesis' way of telling the test failed?